utils/db-generator/query_db_generator.cpp

```diff
@@ -0,0 +1,1176 @@
+    int32_t day = rand() % 12 + 1;
+    int32_t hours = rand() % 24;
+    int32_t minutes = rand() % 60;
+    int32_t seconds = rand() % 60;
+    char ans[22];
+    sprintf(
+            ans,
+            "'%04u-%02u-%02u %02u:%02u:%02u'",
+            year,
+            month,
+            day,
+            hours,
+            minutes,
+            seconds);
+    return std::string(ans);
+}
+table_a_column get_table_a_column(const std::string & c)
+{
+    auto point_place = c.rfind('.');
+    std::string db = "";
+    std::string column = "";
+    if (point_place != std::string::npos)
+    {
+        db = c.substr(0, point_place);
+        column = c.substr(point_place + 1);
+    }
+    else
+    {
+        column = c;
+    }
+    return { db, column };
+}
+
+
+enum type : column_type
+{
+    i = 1,
+    // int
+    f = 2,
+    // float
+    s = 4,
+    // string
+    d = 8,
+    // date
+    dt = 16,
+    // datetime
+    b = 32,
+    // bool
+    all = 63,
+    a = 64,
+    // array
+};
+
+
+std::map<column_type, std::string> type_definition = {
+        {type::i, "Int64"}, {type::f, "Float64"}, {type::s, "String"}, {type::d, "Date"}, {type::dt, "DateTime"}, {type::b, "UInt8"}
+};
+column_type time_type(std::string value)
+{
+    if (value.length() == 12)
+    {
+        if (value[0] != '\'')
+            goto not_date;
+        if (value[11] != '\'')
+            goto not_date;
+        for (int i = 1; i < 5; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[5] != '-')
+            goto not_date;
+        for (int i = 6; i < 8; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[8] != '-')
+            goto not_date;
+        for (int i = 9; i < 11; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        return type::d;
+    }
+
+    if (value.length() == 21)
+    {
+        if (value[0] != '\'')
+            goto not_date;
+        if (value[20] != '\'')
+            goto not_date;
+        for (int i = 1; i < 5; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[5] != '-')
+            goto not_date;
+        for (int i = 6; i < 8; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[8] != '-')
+            goto not_date;
+        for (int i = 9; i < 11; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[11] != ' ')
+            goto not_date;
+        for (int i = 12; i < 14; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[14] != ':')
+            goto not_date;
+        for (int i = 15; i < 17; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        if (value[17] != ':')
+            goto not_date;
+        for (int i = 18; i < 20; ++i)
+            if (!isdigit(value[i]))
+                goto not_date;
+        return type::dt;
```

Comment:
Seriosly

Reply:
I agree, i should refactor this func.
I understand that it looks like crutch.
Is there a better way to define date or datetime about string?